chg: creator e-mail in the event details, fixes #1252

cristianbell opened the pull request:
#### What does it do?

It changes showing the API key behavior, not shown for users where it is not enabled. 
#### Questions
- [ ] Does it require a DB change?
- [ ] Are you using it in production?
- [ ] Does it require a change in the API (PyMISP for example)?
#### Release Type:
- [ ] Major
- [X] Minor
- [ ] Patch

cristianbell committed:
chg: creator e-mail in the event details, fixes #1252
previous behaviour:
- if the logged in user is a site admin, fetch also the event creator details (id, email address)
new behaviour:
- if the logged in user is a site admin OR org admin, fetch also the event creator details.
- check if the event org_id OR orgc_id are the same as the logged in user. IF they are not AND the user is just org admin, strip the event creator details.

rotanid commented on app/Model/Event.php:
```diff
@@ -1293,10 +1293,14 @@ public function fetchEvent($user, $options = array()) {
 			)
 		);
-		if ($user['Role']['perm_site_admin']) {
+		if ($user['Role']['perm_site_admin'] | $user['Role']['perm_admin']) {
```
i think this has to be "||", double not only once

rotanid replied:
was it intentional @cristianbell that you removed this change completely instead of fixing it?

iglocska replied:
Yeah, we had a chat about it.

cristianbell committed:
chg: creator e-mail in the event details, fixes #1252
- fetching the event creator info in a different way, using getEventCreatorInfo();

cristianbell committed:
chg: creator e-mail in the event details, fixes #1252
- removed unnecessary check
- optimized checks
- moved getCreatorInfo function to User Model.
